test/cql-pytest/test_native_transport.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# The cql-pytest framework is about testing CQL functionality, so
+# implementation details like sstables cannot be tested directly. However,
+# we are still able to reproduce some bugs by tricks such as writing some
+# data to the table and then force it to be written to the disk (nodetool
+# flush) and then trying to read the data again, knowing it must come from
+# the disk.
+#############################################################################
+
+import pytest
+import time
+import nodetool
+from util import new_test_table, new_cql
+from cassandra.cluster import NoHostAvailable
+
+def test_enable_disable_binary(cql, test_keyspace):
+    schema = 'k text, v text, primary key (k)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"INSERT INTO {table} (k,v) VALUES ('foo', 'bar')")
+        cql.execute(f"SELECT * FROM {table}")
+
+        nodetool.disablebinary(cql)
+        select_success = False
+        try:
+            cql.execute(f"SELECT * FROM {table}")
+            select_success = True
+        except NoHostAvailable:
+            pass
+        assert not select_success
+
+        nodetool.enablebinary(cql)
+        select_success = False
+        for i in range(5):
+            try:
+                with new_cql(cql) as ncql:
+                    print('Query 3')
+                    ncql.execute(f"SELECT * FROM {table}")
+                    select_success = True
+                    break
+            except NoHostAvailable:
+                time.sleep(1)
```

Comment:
Why sleep a whole second (making this test take a whole second, and are you sure that a 5 second timeout? I would sleep a tiny amount of time (e.g., 0.1 seconds) and have a huge timeout (e.g., 100 seconds), to make the test faster and on the other hand - prevent surprises on very slow machines.